### PR TITLE
Fixed tiaConfig of undefined error. 

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
Link to mseng [bug](https://mseng.visualstudio.com/VSOnline/Automated%20Testing/_queries?id=1093123&_a=edit&triage=true)

### CODE REVIEW

File: Tasks/VsTest/taskinputparser.ts [hunk 1]
```diff
@@ -186,7 +187,12 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
 
     testConfiguration.taskInstanceIdentifier = uuid.v1();
 
-    versionFinder.getVsTestRunnerDetails(testConfiguration);
+    try {
+        versionFinder.getVsTestRunnerDetails(testConfiguration);
+    } catch (error) {
+        utils.Helper.publishEventToCi(AreaCodes.SPECIFIEDVSVERSIONNOTFOUND, error.message, 1039, true);
+        tl.setResult(tl.TaskResult.Failed, error);
```
Comment (prawalagarwal) [inline Tasks/VsTest/taskinputparser.ts]: Dont set result here. Throw here and catch where this function is called (i.e. main vstest.ts). We should not set result at any more places now

Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: okay

File: Tasks/VsTest/taskinputparser.ts [hunk 2]
```diff
@@ -186,7 +187,12 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
 
     testConfiguration.taskInstanceIdentifier = uuid.v1();
 
-    versionFinder.getVsTestRunnerDetails(testConfiguration);
+    try {
+        versionFinder.getVsTestRunnerDetails(testConfiguration);
+    } catch (error) {
+        utils.Helper.publishEventToCi(AreaCodes.SPECIFIEDVSVERSIONNOTFOUND, error.message, 1039, true);
+        throw error;
```
Comment (prawalagarwal) [inline Tasks/VsTest/taskinputparser.ts]: make sure while testing that this is propagated to the final catch

Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: validated

**Comment by RobertoPrevato** [comment]
@prawalagarwal @ShreyasRmsft 
Hi,
I just got this exception, so it seems the bug returned? Maybe after some re-deployment, to fix the VSTS 
service disruption that happened two days ago? (10/10/2017).

**Pull request merged**